### PR TITLE
Don't pipe script and misc. CircleCI fixes

### LINKED ISSUE #57: (CircleCI) sh: 117: Syntax error: end of file unexpected (expecting "fi")
Failing build: https://circleci.com/gh/viddo/atom-textual-velocity/3

[atom/ci/build-package.sh#L117 is the eslint clause](https://github.com/atom/ci/blob/28b4860255fd2c2a25f98b3499c02c702b8160c6/build-package.sh#L117), but can't see any obvious error there., just to rule that out also tried to removed all eslint-related stuff, but it still fails with same error output, see https://circleci.com/gh/viddo/atom-textual-velocity/5

Tried to rebuild using SSH enabled, so I could retry manually on the machine, but didn't manage to reproduce it. First time it stops due to not finding the specs:
```bash
ubuntu@box158:~$ curl -s https://raw.githubusercontent.com/atom/ci/master/build-package.sh | sh
…
Installing modules ✓
Missing spec folder! Please consider adding a test suite in './spec'
```

So `cd` into the dir that contains the source, re-running the build script there the specs run:
```bash
# cd to the existing dir which seems to contain the checked out code and rerun the build script there the tests work…:
ubuntu@box158:~$ cd atom-textual-velocity/
ubuntu@box158:~/atom-textual-velocity$ ls
appveyor.yml  atom-amd64.deb  CHANGELOG.md  circle.yml 	CONTRIBUTING.md  decls 	docs  keymaps  lib  LICENSE.md 	node_modules  package.json  README.md  spec  styles
ubuntu@box158:~/atom-textual-velocity$ curl -s https://raw.githubusercontent.com/atom/ci/master/build-package.sh | sh
Downloading latest Atom release...
…
Installing modules ✓
Running specs...
[23239:0920/104912:ERROR:browser_main_loop.cc(217)] Running without the SUID sandbox! See https://chromium.googlesource.com/chromium/src/+/master/docs/linux_suid_sandbox_development.md for more information on developing with the sandbox on.
Xlib:  extension "RANDR" missing on display ":99".
Xlib:  extension "RANDR" missing on display ":99".
[23239:0920/104919:INFO:CONSOLE(52)] "Window load time: 5960ms", source: file:///usr/share/atom/resources/app.asar/static/index.js (52)
......
```

Any ideas of what might be wrong or suggestions on how to troubleshoot this further?

### PR DESCRIPTION
**Pull request by Arcanemagus**
* Stop piping the script through to `sh`, if the connection is closed mid-execution the script becomes corrupt. See [here](https://www.seancassidy.me/dont-pipe-to-your-shell.html) for some examples.
* Define `NPM_SCRIPT_PATH` on CircleCI so linting runs on there again.
* Remove `local` from the `has_linter` function, this isn't available in POSIX `sh`.

Fixes #57.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,7 +15,10 @@ os:
   - osx
 
 ### Generic setup follows ###
-script: 'curl -s https://raw.githubusercontent.com/atom/ci/master/build-package.sh | sh'
+script:
+  - curl -s https://raw.githubusercontent.com/atom/ci/master/build-package.sh > build-package.sh
```
Comment (joefitzgerald) [inline .travis.yml]: `curl -s -O https://raw.githubusercontent.com/atom/ci/master/build-package.sh`

File: circle.yml [hunk 2]
```diff
@@ -1,6 +1,8 @@
 test:
   override:
-    - curl -s https://raw.githubusercontent.com/atom/ci/master/build-package.sh | sh
+    - curl -s https://raw.githubusercontent.com/atom/ci/master/build-package.sh > build-package.sh
```
Comment (joefitzgerald) [inline circle.yml]: `curl -s -O https://raw.githubusercontent.com/atom/ci/master/build-package.sh`

File: build-package.sh [hunk 3]
```diff
@@ -94,7 +95,7 @@ if [ -n "${APM_TEST_PACKAGES}" ]; then
 fi
 
 has_linter() {
-  local result="$( ${NPM_SCRIPT_PATH} ls --parseable --dev --depth=0 "$1" 2> /dev/null )"
+  result="$( ${NPM_SCRIPT_PATH} ls --parseable --dev --depth=0 "$1" 2> /dev/null )"
```
Comment (joefitzgerald) [inline build-package.sh]: Perhaps make result a bit more obscure? `linter_does_exist`?

**Comment by Arcanemagus** [comment]
Looks like [CircleCI](https://circleci.com/gh/AtomLinter/linter-stylelint/555) [Ubuntu 14.04] and [Travis-CI](https://travis-ci.org/AtomLinter/linter-stylelint/builds/161430971) [Ubuntu 12.04, OS X] worked fine with this change. (And AppVeyor, but this doesn't affect that.)

**Pull request merged**